pkg/backends/otlp/backend.go

```diff
@@ -0,0 +1,237 @@
+								uint64(t.Timestamp),
+								data.WithNumberDataPointMap(btags),
+								data.WithNumberDatapointIntValue(int64(count)),
+							)),
+						),
+					)
+				}
+				return
+			}
+			calcs := []struct {
+				discarded bool
+				suffix    string
+				value     func(data.NumberDataPoint)
+			}{
+				{discarded: bd.discarded.Lower, suffix: "lower", value: data.WithNumberDataPointDoubleValue(t.Min)},
+				{discarded: bd.discarded.Upper, suffix: "upper", value: data.WithNumberDataPointDoubleValue(t.Max)},
+				{discarded: bd.discarded.Count, suffix: "count", value: data.WithNumberDatapointIntValue(int64(t.Count))},
+				{discarded: bd.discarded.CountPerSecond, suffix: "count_ps", value: data.WithNumberDataPointDoubleValue(t.PerSecond)},
+				{discarded: bd.discarded.Mean, suffix: "mean", value: data.WithNumberDataPointDoubleValue(t.Mean)},
+				{discarded: bd.discarded.Median, suffix: "median", value: data.WithNumberDataPointDoubleValue(t.Median)},
+				{discarded: bd.discarded.StdDev, suffix: "std", value: data.WithNumberDataPointDoubleValue(t.StdDev)},
+				{discarded: bd.discarded.Sum, suffix: "sum", value: data.WithNumberDataPointDoubleValue(t.Sum)},
+				{discarded: bd.discarded.SumSquares, suffix: "sum_squares", value: data.WithNumberDataPointDoubleValue(t.SumSquares)},
+			}
+
+			for _, calc := range calcs {
+				if calc.discarded {
+					continue
+				}
+				group.Insert(
+					bd.is,
+					resources,
+					data.NewMetric(fmt.Sprintf("%s.%s", name, calc.suffix)).SetGauge(
+						data.NewGauge(data.NewNumberDataPoint(
+							uint64(t.Timestamp),
+							data.WithNumberDataPointMap(attributes),
+							calc.value,
+						)),
+					),
+				)
+			}
+
+			for _, pct := range t.Percentiles {
+				group.Insert(bd.is, resources, data.NewMetric(fmt.Sprintf("%s.%s", name, pct.Str)).SetGauge(
+					data.NewGauge(data.NewNumberDataPoint(
+						uint64(t.Timestamp),
+						data.WithNumberDataPointMap(attributes),
+						data.WithNumberDataPointDoubleValue(pct.Float),
+					)),
+				))
+			}
+
+		case false:
+			// Computing Timers as Histograms
+			opts := []func(data.HistogramDataPoint){
+				data.WithHistogramDataPointStatistics(t.Values),
+				data.WithHistogramDataPointAttributes(attributes),
+			}
+			if len(t.Histogram) != 0 {
+				opts = append(opts, data.WithHistogramDataPointBucketValues(t.Histogram))
+			}
+			group.Insert(bd.is, resources, data.NewMetric(name).SetHistogram(
+				data.NewHistogram(data.NewHistogramDataPoint(uint64(t.Timestamp), opts...)),
+			))
+		}
+
+	})
+
+	if err := bd.postMetrics(ctx, group.Values()); err != nil {
+		bd.logger.WithError(err).WithFields(logrus.Fields{
+			"endpoint": bd.endpoint,
+		}).Error("Issues trying to submit data")
+		cb(multierr.Errors(err))
+	}
+}
+
+func (c *Backend) postMetrics(ctx context.Context, resourceMetrics []data.ResourceMetrics) error {
+	req, err := data.NewMetricsRequest(ctx, c.endpoint)
+	if err != nil {
+		return err
+	}
+	resp, err := c.client.Do(req)
```

Comment:
IIRC we don't handle the status codes in ProcessMetricResponse so we should handle it here